src/lib.rs

```diff
@@ -167,3 +167,55 @@ mod parsing {
         }
     }
 }
+
+#[cfg(feature = "parsing")]
+pub mod parsing {
+    //! This module contains a set of exported nom parsers which can be used to
+    //! build nom parsers for custom grammars when used alongside the `syn_nom`
+    //! crate.
+    //!
+    //! `syn` uses its own custom fork of `nom`, `syn_nom`, which is smaller, and
+    //! thus improves build speeds. This should be used instead of `nom` when
+    //! building parsers using these parsers.
+
+    pub use macro_input::parsing::macro_input;
+
+    #[cfg(feature = "full")]
+    pub use krate::parsing::krate;
+
+    #[cfg(feature = "full")]
+    pub use item::parsing::item;
+
+    #[cfg(feature = "full")]
+    pub use item::parsing::items;
+
+    #[cfg(feature = "full")]
+    pub use expr::parsing::expr;
+
+    pub use lit::parsing::lit;
+
+    pub use lit::parsing::string as str_lit;
+
+    pub use lit::parsing::byte_string as byte_str_lit;
+
+    pub use lit::parsing::byte as byte_lit;
+
+    pub use lit::parsing::character as char_lit;
+
+    pub use lit::parsing::float as float_lit;
+
+    pub use lit::parsing::int as int_lit;
+
+    pub use lit::parsing::boolean as bool_lit;
```

Comment:
I would prefer to have these lit parsers return their own individual types - it seems silly to parse a bool_lit and then have to pattern-match on a Lit to read the bool out of it. There should be a parser called "boolean" that gives you a `bool`.

For string / byte string / int / float we will need some sort of wrapper struct like `struct StrLit(String, StrStyle)`.